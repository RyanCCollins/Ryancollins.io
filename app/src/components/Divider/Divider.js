import React, { PropTypes } from 'react';
import './Divider.scss';

const Divider = ({
  color
}) => (
  <span
    className={color ? `divider divider__${color}` : 'divider'}
  />
);

Divider.propTypes = {
  color: PropTypes.string
};

export default Divider;
