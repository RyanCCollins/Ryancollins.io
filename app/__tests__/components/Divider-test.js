import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Divider from '../../src/components/Divider/Divider';

// needed because of this:
// https://github.com/facebook/jest/issues/1353
jest.mock('react-dom');

describe('<Divider />', () => {
  it('renders with default props', () => {
    const component = renderer.create(
      <Divider />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders with a color set', () => {
    const component = renderer.create(
      <Divider color={'#123456'} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
