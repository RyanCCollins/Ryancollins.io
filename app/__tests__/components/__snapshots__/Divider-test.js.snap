exports[`<Divider /> renders with a color set 1`] = `
<span
  className="divider divider__blue" />
`;

exports[`<Divider /> renders with default props 1`] = `
<span
  className="divider" />
`;
